src/scripts/main.js

```diff
@@ -1,3 +1,124 @@
+/* eslint-disable no-shadow */
 'use strict';
 
-// write code here
+document.querySelectorAll('thead th').forEach((th, index) => {
+  th.addEventListener('click', () => {
+    const table = th.closest('table');
+    const rows = Array.from(table.querySelectorAll('tbody tr'));
+    const asc = !th.classList.contains('asc');
+
+    th.classList.toggle('asc', asc);
+    th.classList.toggle('desc', !asc);
+
+    rows.sort((a, b) => {
+      const aText = a.cells[index].innerText.trim();
+      const bText = b.cells[index].innerText.trim();
+
+      return asc
+        ? aText.localeCompare(bText, undefined, { numeric: true })
+        : bText.localeCompare(aText, undefined, { numeric: true });
+    });
+
+    rows.forEach((row) => table.querySelector('tbody').appendChild(row));
+  });
+});
+
+document.querySelectorAll('tbody tr').forEach((row) => {
+  row.addEventListener('click', () => {
+    document
+      .querySelectorAll('tbody tr')
+      .forEach((r) => r.classList.remove('active'));
+  });
+});
+
+const form = document.createElement('form');
+
+form.className = 'new-employee-form';
+
+form.innerHTML = `
+  <label>Name: <input name="name" type="text" data-qa="name" required></label>
+  <label>Position: <input name="position" type="text" data-qa="position" required></label>
+  <label>Office:
+    <select name="office" data-qa="office">
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select>
+  </label>
+  <label>Age: <input name="age" type="number" min="18" max="90" data-qa="age" required></label>
+  <label>Salary: <input name="salary" type="number" data-qa="salary" required></label>
+  <button type="submit">Save to table</button>
+`;
+document.body.appendChild(form);
+
+form.addEventListener('submit', (even) => {
+  even.preventDefault();
+
+  const name = form.name.value.trim();
+  const position = form.name.value.trim();
```

Comment:
There is a mistake here: the 'position' variable is incorrectly assigned the value of 'form.name.value'. It should be 'form.position.value'.